File: BasicClassValidation/Classes/Book.vb
Imports System.ComponentModel.DataAnnotations
Imports BasicClassValidation.Enumerations
Imports BasicClassValidation.Rules

Namespace Classes
    Public Class Book
        <Required(ErrorMessage:="{0} is required")>
        Public Property Title() As String

        <Required(ErrorMessage:="{0} is required")>
        Public Property ISBN() As String

        <RequiredEnum(ErrorMessage:="{0} is required.")>
        Public Property Category() As BookCategory

        <ListHasElements(ErrorMessage:="{0} must contain at lease one note")>
        Public Property NotesList() As List(Of String)

    End Class
End Namespace

File: BaseModelsLibrary/Models/Book.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BaseDataValidatorLibrary.CommonRules;

namespace BaseModelsLibrary.Models
{
    public class Book
    {
        [Required(ErrorMessage = "{0} is required")]
        public string Title { get; set; }

        [Required(ErrorMessage = "{0} is required")]
        public string ISBN { get; set; }

        [RequiredEnum(ErrorMessage = "{0} is required")]
        public BookCategory? Category { get; set; }

        [ListHasElements(ErrorMessage = "{0} must contain at least one note")]
        public List<string> NotesList { get; set; }

    }
}
